Lib/test/test_locale_coercion.py

```diff
@@ -0,0 +1,199 @@
+# Tests the attempted automatic coercion of the C locale to a UTF-8 locale
+
+import unittest
+import os
+import sys
+import sysconfig
+import shutil
+import subprocess
+import test.support
+from test.support.script_helper import (
+    run_python_until_end,
+    interpreter_requires_environment,
+)
+
+# In order to get the warning messages to match up as expected, the candidate
+# order here must much the target locale order in Programs/python.c
+_C_UTF8_LOCALES = (
+    # Entries: (Target locale, target category, expected env var updates)
+    ("C.UTF-8", "LC_ALL", "LC_ALL & LANG"),
+    ("C.utf8", "LC_ALL", "LC_ALL & LANG"),
+    ("UTF-8", "LC_CTYPE", "LC_CTYPE"),
+)
+
+# There's no reliable cross-platform way of checking locale alias
+# lists, so the only way of knowing which of these locales will work
+# is to try them with locale.setlocale(). We do that in a subprocess
+# to avoid altering the locale of the test runner.
+def _set_locale_in_subprocess(locale_name, category):
+    cmd_fmt = "import locale; print(locale.setlocale(locale.{}, '{}'))"
+    cmd = cmd_fmt.format(category, locale_name)
+    result, py_cmd = run_python_until_end("-c", cmd, __isolated=True)
+    return result.rc == 0
+
+# Details of the CLI warning emitted at runtime
+CLI_COERCION_WARNING_FMT = (
+    "Python detected LC_CTYPE=C: {} coerced to {} (set another locale "
+    "or PYTHONCOERCECLOCALE=0 to disable this locale coercion behaviour)."
```

Comment:
Docs fixed in https://github.com/python/cpython/pull/659/commits/1c3a2706ef276ca66f676218395b9d792ef3ca0e